### DISCUSSION TITLE
HIP-198: Ledger ID

### DISCUSSION THREAD

**Discussion by sergmetelin** [hiero-ledger/hiero-improvement-proposals#201] [Edits]
This thread is to discuss HIP-198: Ledger ID

**Comment by sergmetelin** [comment]
This message is a `Last Call` notification calling for community feedback on this HIP.

If no substantial changes will be requested for this HIP before `2021-11-22 07:00:00 UTC`, this HIP will change its status to `Council Review`.

**Comment by HederaGraph** [comment]
PROPOSAL FOR PLANNING THE SEGMENTATION AND NUMBER OF CHARACTERS FOR WALLET CODES RELATED TO IDENTITY

Hedera's wallet token ID has a structure that is far superior to wallet codes on all other platforms
Displaying the wallet code in numbers is a huge advantage and benefit, because the display of the hederea wallet code will easily meet all identity requirements in all industries
- The structure of the hedera wallet code can easily be used for identification as a citizen identification number in countries, it is similar to a bank account, and can also be used as a mobile telecommunications system... and in most areas of life related to human identity

Based on these advantages, I would like to propose a proposal for the allocation of numbers according to the structure of the wallet code hedera <shardNum>.<realmNum>.<account> . as follows
1The initial planning with the <account> section will be limited by default to less than 15 numeric characters to apply to identity issues, or ID numbers, cbdc bank account numbers...., if the number exceeds 15 numeric characters, the segments in the <realmNum> section will continue to be developed
and with this approach, all tokens with a format of 15 numeric characters or more up to the end of 8 numeric bytes in the <account> section will be assigned a default token number for nfts, types of messenger topics....

As I observe at the present time, with typical examples such as IKAD learning certificate nfts, or for example Topic Messages
all exploit tokens or many types of nfts (junk, miscellaneous) are assigned to bank tokens with the user's Wallet token ID
Using resources like this is a waste, and in the future, there will be countless products mass production is assigned id tokens in such a wasteful way
On the other hand, if we know how to take advantage of and plan this problem, the power of hedera will be closer to real life, and will increase the value of hedera tokens many times over other block change wallet codes.
Imagine, hedera's ID token wallet will act as a cross-chain bridge, and users will transfer money to each other as conveniently as sending bank account numbers to each other in real life. Even tokens of the same type (for example ETH) will be very reluctant to transfer directly on the chain to each other, but instead, they will transfer through the hedera bridge with a very small fee in exchange for convenience and ease
Therefore, plan from the beginning, do not let wallet codes, identity accounts have too long characters.... At the same time, you should also reconsider how to assign nfts, Topic Messages... to long character wallet codes, in order to optimize IDtoken resources with numbers short characters for identity needs and user wallets

Please share and interact, send this suggestions and ideas to hedera developers
